### DISCUSSION TITLE
Is there plan to add Altair Visualizations suport?

### DISCUSSION THREAD

**Discussion by murilomm192** [zauberzeug/nicegui#657] [Ideas / Feature Requests]
Hello,

I'm playing with NiceGUI and it's great.

In my workflow I use a lot the charts and my go to library is Altair, super easy to use thanks to the Vega API and declarative aproach to making graphs.

is there plans to include Altair support like there is for Plotly?

I Tried using a hack passing a HTML representation of the to the ui.add_body_html but the tags in the graph conflict with the page, creating some unintended behaivoir like the graph show bellow other ui elements like the exemple bellow.

![image](https://user-images.githubusercontent.com/9472259/228287661-0e3611f7-5980-4260-9738-4b1dd066ea41.png)

**Comment by rodja** [comment]
We do not have a `ui.altair(...)` element yet. But it would be a very nice addition to `ui.chart` and `ui.plotly`. Would you like to add it? We need all the help there is to expand the NiceGUI universe. See our [docs for detailed instructions to start contributing](https://github.com/zauberzeug/nicegui/blob/main/CONTRIBUTING.md#new-elements).

**Reply by murilomm192** [reply]
I'll look into it. would be a nice addition. thanks

**Reply by Jerold-S** [reply]
@murilomm192 did you ever make any progress on this?
Just out of question, how did you get the hack you show above to work?

**Reply by murilomm192** [reply]
not really.

I tried, but at the time I had no experience in JS or frontend in general and I did not get very far.

The altair library use vega under the hood, so the hack was getting the html from the generated chart and passing that to niceGUI.

**Reply by azjps** [reply]
It would be pretty sweet if `nicegui` were to support [`anywidget`](https://anywidget.dev/), it might help make altair integration "for free" via `altair.JupyterChart` https://altair-viz.github.io/user_guide/interactions/jupyter_chart.html#supported-environments (along with anything else built on `anywidget` like https://github.com/manzt/quak)

**Reply by falkoschindler** [reply]
@azjps Interesting! I've never heard of anywidget. But it sounds like a great way to add support for other third-party elements. Feel free to create a separate feature request so we can discuss it in more detail and see how an implementation could look like.

**Reply by azjps** [reply]
@falkoschindler sure happy to do so, my understanding is pretty cursory, but I created a new feature request here: https://github.com/zauberzeug/nicegui/discussions/5096.

**Comment by falkoschindler** [comment]
`ui.altair` has just been released via PR #5137 in NiceGUI 3.5.0.
